docs/guides/1238-making-reusable-package.md

```diff
@@ -0,0 +1,188 @@
+```
+
+- At the root of this new folder, create the Cue file that will contain the `#Run` package. The name of the file is not important, but the package name shall follow the name of the remote repo (best practice convention):
+
+```cue title=personal/main.cue
+package personal
+
+import(
+  "universe.dagger.io/alpine"
+  "universe.dagger.io/bash"
+)
+
+#Run: {
+    _img: alpine.#Build & {
+        packages: bash: _
+    }
+
+    bash.#Run & {
+        always: true
+        input:  _img.output
+    }
+}
+```
+
+You should now have below file and directory structure:
+
+```console
+$ tree -L 2
+.
+├── personal
+│   └── main.cue
+└── rootProject
+    ├── cue.mod
+    └── main.cue
+
+3 directories, 2 files
+```
+
+### Link package to the project
+
+- We first need to include the newly created package in the project. In order to do that, we will create a symlink similar to what `dagger project update` would create once we push the package on Github:
+
+```console
+$ ls -l
+total 0
+drwxr-xr-x  4 home  wheel  128  9 mai 16:04 personal
+drwxr-xr-x  4 home  wheel  128  9 mai 16:01 rootProject
+
+$ mkdir -p rootProject/cue.mod/pkg/github.com/your-username/
+
+$ ln -s "$(pwd)/personal" "$(pwd)/rootProject/cue.mod/pkg/github.com/your-username/personal"
+```
+
+When using your package from `dagger project update` in the `rootProject` directory, the actual packager manager would copy the files from the repository in the `rootProject/cue.mod/pkg/github.com/your-username/personal` folder.
+
+- We then need to change the project's `main.cue` to call the `#Run` definition in the `personal` package that we just built:
+
+```cue title="rootProject/main.cue"
+package main
+
+import (
+    "dagger.io/dagger"
+
+    "github.com/your-username/personal" // import personal package
+)
+
+dagger.#Plan & {
+    actions: {
+        bar: personal.#Run & { // reference #Run definition from personal package imported above
+            script: contents: "echo \"Hello!\""
+        }
+    }
+}
+```
+
+### Run the project
+
+Now that we have connected all the dots, let's run our plan to see if it works:
+
+```console
+$ cd rootProject
+
+$ dagger do bar --log-format plain
+4:42PM INF actions.bar.script._write | computing
+4:42PM INF actions.bar._img._dag."0"._op | computing
+4:42PM INF actions.bar.script._write | completed    duration=0s
+4:42PM INF actions.bar._img._dag."0"._op | completed    duration=0s
+4:42PM INF actions.bar._img._dag."1"._exec | computing
+4:42PM INF actions.bar._img._dag."1"._exec | completed    duration=0s
+4:42PM INF actions.bar._exec | computing
+4:42PM INF actions.bar._exec | completed    duration=200ms
+4:42PM INF actions.bar._exec | #5 0.123 Hello!
+```
+
+### Push package on repository
+
+Now that we made sure we correctly built our package, we only need to push it to the repository:
+
+- Add
+- Commit
+- Tag
+- Push
+
+On another project, you will directly be able to retrieve the `personal` package using the `dagger project update url@tag` command.
```

Comment:
That would be awesome to provide a real url example.
For instance, `https://github.com/dagger/dagger/ci` works

Reply:
I'd agree more if we had a simpler, more reusable example.

Reply:
I'd agree more if we had a simpler, more reusable example.